File: src/main/java/io/bindernews/thegrackle/cards/Defend_GK.java
package io.bindernews.thegrackle.cards;

import basemod.AutoAdd;
import com.megacrit.cardcrawl.actions.common.GainBlockAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import io.bindernews.bnsts.CardNums;

@AutoAdd.Seen
public class Defend_GK extends BaseCard {
    public static final CardConfig C = new CardConfig("Defend_GK", CardType.SKILL);
    public static final CardNums NUM = CardNums.builder()
            .cost(1)
            .block(5).blockUpg(8)
            .build();

    public Defend_GK() {
        super(C, CardRarity.BASIC, CardTarget.SELF);
        tags.add(CardTags.STARTER_DEFEND);
        NUM.init(this);
    }

    @Override
    public void apply(AbstractCreature p, AbstractCreature m) {
        addToBot(new GainBlockAction(p, p, block));
    }

    @Override
    public void upgrade() {
        NUM.upgrade(this, false);
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/cards/Defend_GK.kt
package io.bindernews.thegrackle.cards

import basemod.AutoAdd.Seen
import com.megacrit.cardcrawl.actions.common.GainBlockAction
import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.thegrackle.helper.CardVariables

@Seen
class Defend_GK : BaseCard(C, VARS) {
    override fun apply(p: AbstractCreature, m: AbstractCreature?) {
        addToBot(GainBlockAction(p, p, block))
    }

    companion object {
        @JvmField
        val C = CardConfig("Defend_GK", CardType.SKILL, CardRarity.BASIC, CardTarget.SELF)
        val VARS = CardVariables.config { c ->
            c.cost(1, -1)
            c.block(5, 8)
            c.tags(CardTags.STARTER_DEFEND)
        }
    }
}